app/controllers/merchant/orders_controller.rb

```diff
@@ -1,6 +1,24 @@
-class Merchant::OrdersController < ApplicationController
+class Merchant::OrdersController < Merchant::BaseController
 
   def show
-  end 
+    @order = Order.find(params[:order_id])
+    @merchant_id = User.find(session[:user_id]).merchant_id
+  end
 
+  def update
+    item_order = ItemOrder.where(item_id: params[:item_id], order_id: params[:order_id]).first
+    item_order.fulfill_status = "fulfilled"
+    item_order.save
+
+    item = Item.find(params[:item_id])
+    item.inventory -= item_order.quantity
+    flash[:notice] = "Item has been fulfilled" if item.save
+
+    order = Order.find(params[:order_id])
+    if order.all_fulfilled?
+      order.status = "packaged"
+      order.save
+    end
```

Comment:
I think this could be refactored so we aren't making db calls & attribute reassignment in the controller.

Reply:
Disregard.